sycl/test-e2e/Graph/finalize_twice.cpp

```diff
@@ -0,0 +1,18 @@
+// REQUIRES: level_zero, gpu
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// Tests calling finalize() more than once on the same command_graph.
+
+#include "graph_common.hpp"
+
+int main() {
+  queue Queue;
+
+  ext::oneapi::experimental::command_graph Graph{Queue.get_context(),
+                                                 Queue.get_device()};
+  auto GraphExec = Graph.finalize();
+  auto GraphExec2 = Graph.finalize();
```

Comment:
It is kind of implied by `RUN` line that the app should exit with `0` and therefore we expect that there was no exceptions. But that is not entirely clear neither from the code, nor from the comments.

Moreover, strictly speaking, I don't think that the code in the test ensures that there were no exceptions thrown in the program.

There is no exception handler set for `queue` and default one is used. Even though the spec [says](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsubsec:exception.nohandler) that default async handler _must_ call `std::terminate` does not guarantee abnormal termination of the program. It calls `std::abort` by default, but who knows how did we configure terminate handler? Moreover, `std::abort` can technically be caught and ignored.

This is a half nit: all those theoretical situations won't happen most likely, but it would be good to somehow express that we expect no exceptions to be thrown: either through explicit use of `try/catch` or at least through comments.

Reply:
This is a good point. I've added `try/catch` around the things being tested in both this test and also `finalize_empty.cpp` which has also been renamed to `empty_graph.cpp` since it is testing both finalizing and submitting an empty graph.